src/library_syscall.js

```diff
@@ -748,12 +748,31 @@ var SyscallsLibrary = {
   __syscall140: function(which, varargs) { // llseek
     var stream = SYSCALLS.getStreamFromFD(), offset_high = SYSCALLS.get(), offset_low = SYSCALLS.get(), result = SYSCALLS.get(), whence = SYSCALLS.get();
 #if SYSCALLS_REQUIRE_FILESYSTEM
+#if LARGE_FILESYSTEM
+    var HIGH_OFFSET = 0x100000000; // 2^32
+    if (offset_low < 0) offset_low += HIGH_OFFSET;
+    var offset = offset_high * HIGH_OFFSET + offset_low;
+
+    var DOUBLE_LIMIT = 0x20000000000000; // 2^53
+    // we also check for equality since DOUBLE_LIMIT + 1 == DOUBLE_LIMIT
+    if (offset <= -DOUBLE_LIMIT || offset >= DOUBLE_LIMIT) {
+#if ASSERTIONS
+      abort('reached accurate limit for doubles');
```

Comment:
Also, I'm unsure if this should abort. I think it's ok to do `warn('..text..')` to log a warning, and let the return of an error happen anyhow.

Or, if we think returning an overflow error is not the right thing to do, we can abort on this and not do that return - that is, continue to do incorrect behavior otherwise - which is ok, as only ASSERTIONS should be catching it.

In other words, I think it's overly complex to have 2 "failure modes", one in ASSERTIONS that aborts, and one that returns and overflow otherwise.

Reply:
I based this off the old code which had a similar thing. If it would overflow a 32-bit int it either aborted if assertains were turned on or returned `EOVERFLOW` otherwise.